rcl/src/rcl/logging_rosout.c

```diff
@@ -87,13 +89,52 @@ rcl_ret_t rcl_logging_rosout_init(const rcl_allocator_t * allocator)
     rcutils_hash_map_init(
       &__logger_map, 2, sizeof(const char *), sizeof(rosout_map_entry_t),
       rcutils_hash_map_string_hash_func, rcutils_hash_map_string_cmp_func, allocator));
+  if (RCL_RET_OK != status) {
+    return status;
+  }
+
+  __sublogger_names = rcutils_get_zero_initialized_array_list();
+  status = rcl_ret_from_rcutils_ret(
+    rcutils_array_list_init(
+      &__sublogger_names, 2, sizeof(const char **), allocator));
+  if (RCL_RET_OK != status) {
+    status = rcl_ret_from_rcutils_ret(rcutils_hash_map_fini(&__logger_map));
+    return status;
```

Comment:
I don't think we want to return the status from `rcutils_hash_map_fini` here.  Instead, we should return the status from above.